extension.js

```diff
@@ -18,54 +18,102 @@
-const { PanelManager } = Me.imports.panelManager;
-const Utils = Me.imports.utils;
-const AppIcons = Me.imports.appIcons;
+import {Extension, gettext as _} from 'resource:///org/gnome/shell/extensions/extension.js';
+
+const Mainloop = imports.mainloop;
+const {signals: Signals} = imports;
 
 const UBUNTU_DOCK_UUID = 'ubuntu-dock@ubuntu.com';
 
 let panelManager;
 let extensionChangedHandler;
 let disabledUbuntuDock;
-let extensionSystem = (Main.extensionManager || imports.ui.extensionSystem);
+let extensionSystem = Main.extensionManager;
+
+export var DTP_EXTENSION = null;
+export var SETTINGS = null;
+export var DESKTOPSETTINGS = null;
+export var PERSISTENTSTORAGE = null;
+export var EXTENSION_UUID = null;
+export var EXTENSION_PATH = null;
+
+export default class DashToPanelExtension extends Extension {
+    constructor(metadata) {
+        super(metadata);
+        console.log(`Initiating ${this.uuid}`);
+
+        this._realHasOverview = Main.sessionMode.hasOverview;
+
+        this.initTranslations();
+        
+        //create an object that persists until gnome-shell is restarted, even if the extension is disabled
+        PERSISTENTSTORAGE = {};
+    }
 
-function init() {
-    this._realHasOverview = Main.sessionMode.hasOverview;
+    enable() {
+        console.log(_('This is a translatable text'));
 
-    ExtensionUtils.initTranslations(Utils.TRANSLATION_DOMAIN);
-    
-    //create an object that persists until gnome-shell is restarted, even if the extension is disabled
-    Me.persistentStorage = {};
-}
+        DTP_EXTENSION = this;
+
+        // The Ubuntu Dock extension might get enabled after this extension
+        extensionChangedHandler = extensionSystem.connect('extension-state-changed', (data, extension) => {
+            if (extension.uuid === UBUNTU_DOCK_UUID && extension.state === 1) {
+                _enable(this);
+            }
+        });
+
+        //create a global object that can emit signals and conveniently expose functionalities to other extensions 
+        global.dashToPanel = {};
+        Signals.addSignalMethods(global.dashToPanel);
+        
+        _enable(this);
+    }
+
+    disable(reset = false) {
+        panelManager.disable();
+
+        DTP_EXTENSION = null;
+        SETTINGS = null;
+        panelManager = null;
 
-function enable() {
-    // The Ubuntu Dock extension might get enabled after this extension
-    extensionChangedHandler = extensionSystem.connect('extension-state-changed', (data, extension) => {
-        if (extension.uuid === UBUNTU_DOCK_UUID && extension.state === 1) {
-            _enable();
+        // Utils.removeKeybinding('open-application-menu');
+        // Utils.addKeybinding(
+        //     'open-application-menu',
+        //     new Gio.Settings({ schema_id: WindowManager.SHELL_KEYBINDINGS_SCHEMA }),
+        //     Main.wm._toggleAppMenu.bind(Main.wm),
+        //     Shell.ActionMode.NORMAL | Shell.ActionMode.POPUP
+        // );
```

Comment:
open-application-menu and _toggleAppMenu are no longer available, do we have to replace this somehow?